tests/python/pants_test/backend/native/tasks/test_cpp_compile.py

```diff
@@ -59,3 +59,24 @@ def test_caching(self):
     # TODO: what is this testing?
     cpp_compile.execute()
     cpp_compile.execute()
+
+  def test_target_level_toolchain_variant_llvm(self):
+    cpp_lib_target = self.make_target(
+      '//:cpp_library',
+      NativeLibrary,
+      toolchain_variant='llvm'
+    )
+
+    task = self.create_task(self.context(target_roots=[cpp_lib_target]))
+    compiler = task.get_compiler(cpp_lib_target)
+    self.assertIn('llvm', compiler.path_entries[0])
```

Comment:
This is fantastic, and might let us remove some of the integration testing which checks for an info-level log to see if the compiler is correct (a severe hack which will be fixed as a whole through #7128). I wasn't sure how to do this testing, thank you for figuring this out.

Reply:
Could you also please modify this test to set:
```python
options={
  'native-build-step': {
    'toolchain_variant': 'gcc',
  },
}
```
but then set `toolchain_variant='llvm'` on the target to see that it is overridden (that `'llvm'` is selected)? I would also prefer that in `test_target_level_toolchain_variant_default()`, we explicitly set the option value as in the snippet instead of relying on the default set in `pants.ini`.

Reply:
added explicit
```
self.set_options_for_scope('native-build-step', toolchain_variant='gnu')
```
before hand